src/components/fs/FS.cpp

```diff
@@ -0,0 +1,194 @@
+#include "FS.h"
+#include <cstring>
+#include <cstdlib>
+#include "drivers/SpiNorFlash.h"
+#include <littlefs/lfs.h>
+#include <lvgl/lvgl.h>
+
+using namespace Pinetime::Controllers;
+
+
+static constexpr size_t BLOCK_SIZE_BYTES = 4096;
+
+/*
+*   Interface between littlefs and SpiNorFlash
+*/
+
+namespace {
+    static int read(const struct lfs_config *c, lfs_block_t block,
+                    lfs_off_t off, void *buffer, lfs_size_t size) {
+        Pinetime::Controllers::FS& lfs = *(reinterpret_cast<Pinetime::Controllers::FS*>(c->context));
+        const size_t address = lfs.mStartAddress + (block * BLOCK_SIZE_BYTES) + off;
+        lfs.mDriver.Read(address, (uint8_t*)buffer, size);
+
+        return 0u;
+    }
+
+    static int prog(const struct lfs_config *c, lfs_block_t block,
+            lfs_off_t off, const void *buffer, lfs_size_t size) {
+        Pinetime::Controllers::FS& lfs = *(reinterpret_cast<Pinetime::Controllers::FS*>(c->context));
+        const size_t address = lfs.mStartAddress + (block * BLOCK_SIZE_BYTES) + off;
+        lfs.mDriver.Write(address, (uint8_t*)buffer, size);
+        return lfs.mDriver.ProgramFailed() ? -1u : 0u;
+    }
+
+    static int erase(const struct lfs_config *c, lfs_block_t block) {
+        Pinetime::Controllers::FS& lfs = *(reinterpret_cast<Pinetime::Controllers::FS*>(c->context));
+        const size_t address = lfs.mStartAddress + (block * BLOCK_SIZE_BYTES);
+        lfs.mDriver.SectorErase(address);
+        return lfs.mDriver.EraseFailed() ? -1u : 0u;
+    }
+
+    static int sync(const struct lfs_config *c) {
+        // no hardware caching used
+        return 0u;
+    }
```

Comment:
Returning unsigned literals in functions returning `int` looks a little weird.

Reply:
Thank you, I will review the recommendations.